### PR TITLE
soc: riscv: andes_v5: Refine soc using syscon driver 

### PR DESCRIPTION
**Pull request by wtlee1995**
Use generic syscon driver to refine andes soc.

Signed-off-by: Wei-Tai Lee [wtlee@andestech.com](mailto:wtlee@andestech.com)

### CODE REVIEW

File: soc/riscv/riscv-privilege/andes_v5/l2_cache.c [hunk 1]
```diff
@@ -81,17 +88,19 @@ static void andes_v5_l2c_enable(void)
 
 static int andes_v5_l2c_init(const struct device *dev)
 {
+	uint32_t system_cfg;
+	const struct device *syscon_dev = DEVICE_DT_GET(DT_NODELABEL(syscon));
```
Comment (carlocaione) [inline soc/riscv/riscv-privilege/andes_v5/l2_cache.c:92]: maybe `const struct device *const ...`?

File: soc/riscv/riscv-privilege/andes_v5/l2_cache.c [hunk 2]
```diff
@@ -81,17 +88,19 @@ static void andes_v5_l2c_enable(void)
...
+
 	ARG_UNUSED(dev);
 
-#ifdef SMU_BASE
-	volatile uint32_t *system_cfg =
-		INT_TO_POINTER(SMU_BASE + SMU_SYSTEMCFG);
+	if (syscon_dev) {
```
Comment (carlocaione) [inline soc/riscv/riscv-privilege/andes_v5/l2_cache.c:96]: please use `device_is_ready` instead

**Review by carlocaione — CHANGES_REQUESTED** [review]

File: dts/riscv/andes/andes_v5_ae350.dtsi [hunk 3]
```diff
@@ -204,8 +204,8 @@
 					       &CPU2_intc 7 &CPU3_intc 7>;
 		};
 
-		smu: smu@f0100000 {
-			compatible = "andestech,atcsmu100";
```
Comment (fkokosinski) [inline dts/riscv/andes/andes_v5_ae350.dtsi:208]: Out of curiosity - is this binding used anywhere else? If not, why are we not removing it?

Comment (wtlee1995) [reply dts/riscv/andes/andes_v5_ae350.dtsi:208]: No, we didn't use this binding and didn't add `andestech,atcsmu100.yaml` either.

File: soc/riscv/riscv-privilege/andes_v5/l2_cache.c (hunk 2 above)
Comment (wtlee1995) [reply soc/riscv/riscv-privilege/andes_v5/l2_cache.c]: Thanks, added it.

**Review by carlocaione — APPROVED** [review]

**Review by fkokosinski — APPROVED** [review]

**Pull request merged**